WordPress/Classes/Extensions/PHAsset+Exporters.swift

```diff
@@ -0,0 +1,281 @@
+            PHImageManager.defaultManager().requestImageForAsset(self, targetSize: requestedSize, contentMode: .AspectFit, options: options) { (image, info) -> Void in
+                guard let image = image
+                else {
+                    if let error = info?[PHImageErrorKey] as? NSError {
+                        errorHandler(error: error)
+                    } else {
+                        errorHandler(error: self.errorForCode(.FailedToExport,
+                            failureReason: NSLocalizedString("Unknown asset export error", comment: "Error reason to display when the export of a image from device library fails")
+                            ))
+                    }
+                    return
+                }
+                do {
+                    try image.writeToURL(url, type: targetUTI, compressionQuality: 0.9, metadata: nil)
+                    successHandler(resultingSize: image.size)
+                } catch let error as NSError {
+                    errorHandler(error: error)
+                }
+            }
+    }
+    
+    var defaultThumbnailUTI: String {
+        get {
+            return kUTTypeJPEG as String
+        }
+    }
+    
+    // MARK: - Error Handling
+    
+    enum ErrorCode : Int {
+        case UnsupportedAssetType = 1
+        case FailedToExport = 2
+        case FailedToExportMetadata = 3
+    }
+    
+    private func errorForCode(errorCode: ErrorCode, failureReason: String) -> NSError {
+        let userInfo = [NSLocalizedFailureReasonErrorKey: failureReason]
+        let error = NSError(domain: "PHAsset+ExporterExtensions", code: errorCode.rawValue, userInfo: userInfo)
+        
+        return error
+    }
+    
+    func requestMetadataWithCompletionBlock(completionBlock:(metadata:[String:AnyObject]) ->(), failureBlock:(error:NSError) -> ()) {
+        let editOptions = PHContentEditingInputRequestOptions();
+        editOptions.networkAccessAllowed = true;
+        self.requestContentEditingInputWithOptions(editOptions) { (contentEditingInput, info) -> Void in
+            guard let contentEditingInput = contentEditingInput,
+                let fullSizeImageURL = contentEditingInput.fullSizeImageURL,
+                let image = CIImage(contentsOfURL:fullSizeImageURL) else {
+                    completionBlock(metadata:[String:AnyObject]())
+                    if let error = info[PHImageErrorKey] as? NSError {
+                        failureBlock(error: error)
+                    } else {
+                        failureBlock(error: self.errorForCode(.FailedToExportMetadata,
+                            failureReason: NSLocalizedString("Unable to export metadata", comment: "Error reason to display when the export of a image from device library fails")
+                            ))
+                    }
+                    return
+            }
+            completionBlock(metadata:image.properties)
+        }
+    }
+    
+    func originalUTI() -> (String?) {
```

Comment:
Code style - I don't think its common to wrap the return type in parenthesis unless its a tuple